ui/src/InvalidAddress.js

```diff
@@ -0,0 +1,107 @@
+import React, {Component} from 'react';
+import '@fontsource/roboto';
+import {Button, Dialog, DialogContent, TextField} from '@material-ui/core';
+
+class InvalidAddress extends Component {
+  state = {
+    reason: '',
+    notes: '',
+    reasonValidationError: false,
+    showDialog: false,
+  }
+
+  openDialog = () => {
+    this.setState({
+      showDialog: true,
+    })
+  }
+
+  closeDialog = () => {
+    this.setState({
+      reason: '',
```

Comment:
do you really need to reset everything when you close, given you reset when you open?

Reply:
I cleared the decks because I wasn't sure if the user would want any text to remain if they reopened the dialog. Happy to keep it there though if it makes more sense to?

Reply:
nah it's fine

Reply:
sorry I didn't see that the open dialog doesn't clear